### ISSUE TITLE
Migrate from old support media to Media2

### ISSUE THREAD

**Issue by TacoTheDank** [enricocid/Music-Player-GO#246]
- https://developer.android.com/jetpack/androidx/releases/media2
- https://stackoverflow.com/questions/57566102/what-are-the-differences-between-mediasessioncompat-and-mediasession-in-android

**Comment by enricocid** [comment]
That was the idea, but I can't find any docs 😵💀
I'm gonna investigate 👍🏻😎

**Comment by TacoTheDank** [comment]
Yeah, for some reason there aren't many things floating around about media2, considering it supersedes the old support media classes in androidx.media. Also it only supports API 21 and up, but that's ok since the minSdk in this project is also 21.

**Comment by MGislv** [comment]
It looks like this https://github.com/h67ma/GnarlyMusicPlayer appears to be using media2, don't know if it helps though

**Comment by enricocid** [comment]
Thanks @MGislv 

Hey @TacoTheDank 
I have a question.

ResourcesCompat.getDrawable... vs AppCompatResources.getDrawable ...

Which one would You choose? I'm confused 😭

**Comment by TacoTheDank** [comment]
@enricocid Oh man, sometimes I don't know myself. Usually I refer to this and try to make up my mind:
- https://stackoverflow.com/questions/43004886/resourcescompat-getdrawable-vs-appcompatresources-getdrawable
(The guy refers to Android versions as API versions for some reason)

Do you actually need a compat class for getDrawable? I thought they weren't necessary if the minimum supported API is 21. :thinking:

**Comment by MGislv** [comment]
def media2_version = "1.1.2"

    // Interacting with MediaSessions
    implementation "androidx.media2:media2-session:$media2_version"
    // optional - UI widgets for VideoView and MediaControlView
    implementation "androidx.media2:media2-widget:$media2_version"
    // optional - Implementation of a SessionPlayer
    implementation "androidx.media2:media2-player:$media2_version"
    //implementation 'androidx.media:media:1.2.1'`

It builds and seems to work, don't know if it changes anything

**Comment by OxygenCobalt** [comment]
I've found [a repository](https://github.com/h6ah4i/Media2Experiment/blob/master/app/src/main/java/com/h6ah4i/example/media2/MainActivity.kt) that seems to show media2 in action. Not sure how helpful it is but I'm trying to extrapolate information to implement media2 in my own player, so maybe this will help?

**Comment by TacoTheDank** [comment]
There's more documentation for media2 in this medium article than there is in the official Android docs https://medium.com/google-exoplayer/the-media2-extension-for-exoplayer-d6b7d89b9063

Media2 has been stable since 2019 and there's no documentation anywhere on it. Android docs are the worst sometimes lol

**Comment by OxygenCobalt** [comment]
And even *that* documentation only goes over the exoplayer-specific extension. There's nothing about uploading your own media items or setting the metadata yourself, and its not like the exoplayer extension is even that good [You can't even return cover images without a blocking call, allofmywhat].

I may as well just read the extension's code at this point and try to figure out how to create a implementation in my own player. If that somehow succeeds I'll hop back to this thread and show how I did it and [maybe] make a pull.

**Comment by TacoTheDank** [comment]
I have found [this cool demo app](https://github.com/h6ah4i/Media2Experiment). Definitely worth checking out to see how it may be applied to your own repos.
